### ISSUE TITLE
LEFT [ ANTI | SEMI ] JOIN ... USING not implemented correctly

### ISSUE THREAD

**Issue by knutwannheden** [jOOQ/jOOQ#9250]
When entering this in the translator:

```sql
select 1
from a left semi join b using (x);
```

The result is:

```sql
select 1
from a
where exists (
  select 1 as one
  from b
  where 1 = 1
)
```

I would have expected a `a.x = b.x` predicate rather than the `1 = 1`.

Same holds true for `LEFT ANTI JOIN`.

**Comment by lukaseder** [comment]
This doesn't seem to be a parser related problem. The SQL generation is simply not implemented correctly. Adding this failing test:

```java
public void testSemiAntiJoinUsing() throws Exception {
    boolean unqualified = unqualifiedUsingColumns();

    Result<Record1<String>> r1 =
    create().select(TAuthor_LAST_NAME())
            .from(TAuthor())
            .leftSemiJoin(TBook())
            .using(TAuthor_ID())
            .orderBy(unqualified
                    ? field("id")
                    : TAuthor_ID())
            .fetch();

    assertEquals(1, r1.size());
    assertEquals(AUTHOR_LAST_NAMES.get(0), r1.get(0).get(TAuthor_LAST_NAME()));

    Result<Record1<String>> r2 =
    create().select(TAuthor_LAST_NAME())
            .from(TAuthor())
            .leftAntiJoin(TBook())
            .using(TAuthor_ID())
            .orderBy(unqualified
                    ? field("id")
                    : TAuthor_ID())
            .fetch();

    assertEquals(1, r2.size());
    assertEquals(AUTHOR_LAST_NAMES.get(1), r2.get(0).get(TAuthor_LAST_NAME()));
}
```

**Comment by lukaseder** [comment]
It's simply not implemented. This is the emulation logic for semi / anti join:

```java
switch (translatedType) {
    case LEFT_SEMI_JOIN:
    case LEFT_ANTI_JOIN:
        if (TRUE.equals(ctx.data(DATA_COLLECT_SEMI_ANTI_JOIN))) {

            @SuppressWarnings("unchecked")
            List<Condition> semiAntiJoinPredicates = (List<Condition>) ctx.data(DATA_COLLECTED_SEMI_ANTI_JOIN);

            if (semiAntiJoinPredicates == null) {
                semiAntiJoinPredicates = new ArrayList<>();
                ctx.data(DATA_COLLECTED_SEMI_ANTI_JOIN, semiAntiJoinPredicates);
            }

            switch (translatedType) {
                case LEFT_SEMI_JOIN:
                    semiAntiJoinPredicates.add(exists(selectOne().from(rhs).where(condition)));
                    break;

                case LEFT_ANTI_JOIN:
                    semiAntiJoinPredicates.add(notExists(selectOne().from(rhs).where(condition)));
                    break;
            }

            return;
        }
}
```

**Comment by lukaseder** [comment]
Fixed in 3.13.0. Backport scheduled for 3.12.2 (#9258)
